components/server/src/workspace/workspace-starter.ts

```diff
@@ -520,7 +520,7 @@ export class WorkspaceStarter {
         spec.setGit(this.createGitSpec(workspace, user));
         spec.setPortsList(ports);
         spec.setInitializer(await initializerPromise);
-        spec.setTheiaVersion(instance.configuration!.theiaVersion);
+        spec.setIdeImage(`${this.env.theiaImageRepo}:${instance.configuration!.theiaVersion}`);
```

Comment:
why not `env.theiaImage`? It's used as fully-qualidfied image name

Reply:
Indeed, the mix is a bit odd. I was trying to stay close to current behaviour where the IDE version is stored at the instance (yet determined by a deploy-time value), whereas the IDE repo just is fixed at deploy-time.

I reckon that in the near future this will be an instance field entirely.